File: 总题库/45.JumpGame2.cpp
// Greedy
class Solution {
public:
    int jump(vector<int>& nums) {
        if (nums.size() < 2) return 0;
        int numsLen = nums.size();
        int border = nums[0], index = 0;
        int stepCounts = 0;
        while (index < numsLen && index <= border) {
            int longest = 0;
            for (int i = index + 1; i <= border; ++i) {
                longest = max(longest, i + nums[i]);
            }
            if (longest > border) {
                border = longest;
                index = border;
                ++stepCounts;
            }
            if (border >= numsLen - 1) return stepCounts;
        }
        return 0;
    }
};

File: 总题库/45.JumpGame2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if len(nums) < 2: return 0
        numsLen, border, index, stepCounts = len(nums), nums[0], 0, 1
        while index < numsLen and index <= border:
            if border >= numsLen - 1:
                return stepCounts
            longest = 0
            for i in range(index + 1, border + 1):
                longest = max(longest, i + nums[i])
            if longest > border:
                index = border
                border = longest
                stepCounts += 1
            else: return 0
        return 0
